lib/appsignal/config.rb

```diff
@@ -395,12 +396,10 @@ def load_from_disk
       return if !config_file || !File.exist?(config_file)
 
       read_options = YAML::VERSION >= "4.0.0" ? { :aliases => true } : {}
-      configurations = YAML.load(ERB.new(IO.read(config_file)).result, **read_options)
+      configurations = YAML.load(ERB.new(File.read(config_file)).result, **read_options)
       config_for_this_env = configurations[env]
       if config_for_this_env
-        config_for_this_env.each_with_object({}) do |(key, value), hash|
-          hash[key.to_sym] = value # convert keys to symbols
-        end
+        config_for_this_env.transform_keys(&:to_sym)
```

Comment:
The `transform_keys` method was added in Ruby 2.5. I think this is okay, since we already consider EOL versions of Ruby to not be supported, and [we removed them from our CI](https://github.com/appsignal/appsignal-ruby/commit/6f9b7a4d12c6ff3353359cb37c5c02af8bbc6ec6#diff-fcf97d076c85af57375095487fc24ed98053f21077be628677d6c45fdeb50322) some time ago, but I'm pointing it out, as this change makes the deprecation of 2.4 and below effective.

Reply:
It's no longer possible to install it on Ruby versions lower than 3.0, since PR https://github.com/appsignal/appsignal-ruby/pull/939